optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/lookup/LookUpManager.java

```diff
@@ -74,4 +75,20 @@ public void clearWorkingObjects() {
         return lookUpStrategy.lookUpWorkingObject(idToWorkingObjectMap, externalObject);
     }
 
+    /**
+     * As defined by {@link ScoreDirector#lookUpWorkingObjectOrReturnNull(Object)}.
+     * @param externalObject sometimes null
+     * @return null if externalObject is null or if there is no workingObject for externalObject
+     * @throws IllegalArgumentException if it cannot be looked up or if the externalObject's class is not supported
+     * @throws IllegalStateException if it cannot be looked up
+     * @param <E> the object type
+     */
+    public <E> E lookUpWorkingObjectOrReturnNull(E externalObject) {
```

Comment:
Given the minimal supported version is Java 8, we should start using Optional instead of returning null value from public methods. Optional makes it very difficult (unfortunately not impossible) to retrieve the value without thinking about corner cases first.